docs/rfcs/template.md

```diff
@@ -0,0 +1,20 @@
+# IMEX Design Document (RFC)
+
+## Summary
+Short description of proposed solution.
+
+## Motivation
+Explain motivation for the proposed idea. 
+If you are proposing a new Dialect, it needs to address below criteria:
```

Comment:
If you are proposing an optimization pass, please explain the applicable scenario and expected performance gain.